adding notes about security on readme file (#1498)
Azure/azure-sdk-for-c: sdk/samples/iot/docs/how_to_iot_hub_esp8266_nodemcu.md

    NOTE: If you are using WSL, do not run these commands from the Windows system drive (e.g. `/mnt/c/`).

    ### Certificates - Important to know

>     For sample purposes, Embedded C SDK samples leverage certificates obtained when executing either `New-ArduinoZipLibrary.ps1` or `generate_arduino_zip_library.sh`. 

Reviewer: ```suggestion
    The Embedded C SDK samples leverage certificates obtained when executing either `New-ArduinoZipLibrary.ps1` or `generate_arduino_zip_library.sh`. These samples also use the Baltimore root CA to validate IoT Hub and Device Provisioning Service (DPS) server certificates.
```

Reviewer 2: I think the message here should be different.
The idea is that the application shall use the certificates for TLS negotiation (server cert validation); 
the scripts being related to it is secondary, they are there just to support the main goal above.

My suggestion:

### Certificates - Important to know
 
The Azure IoT Hub certificates presented during TLS negotiation shall be always validated using the appropriate root CA certificate(s).
For the Node MCU ESP8266 sample our script `generate_arduino_zip_library.sh` automatically downloads the root certificate used in the United States regions (Baltimore CA certificate) and add it to the sketch project.
For other regions (and private cloud environments), please use the appropriate root CA certificate.

>     For sample purposes, Embedded C SDK samples leverage certificates obtained when executing either `New-ArduinoZipLibrary.ps1` or `generate_arduino_zip_library.sh`. 

Reviewer 3: AFAIK we don't pull the certificate when calling the Powershell script (yet). That is a bug reported by @ewertons . 
We can either choose to always require WSL on Windows (.sh script only) or come up with a way to download then transform the certificate into something loadable into the Arduino project (it might be possible to directly load a PEM file?). 

Currently the SH script is using `xxd` not available by default on Windows (easiest way to obtain it on Windows is to install the VIM editor).

Author: you're correct. just changed that to represent the current status.

>     When creating solutions based on the Embedded C SDK you should consider :

Reviewer: ```suggestion
    When creating solutions based on the Embedded C SDK that use certificates, you should consider the following:
```

Reviewer 2: I don't think this is giving the right idea.
There are two types of certs every application (including samples) can work with.
- Root CAs used to validate the server cert. These shall always be employed by applications. We do not recommend people to skip server cert validation.
- Client x509 certs used for device authentication; these are optional (although also recommended) since customers can use SAS tokens.

Author: applied the suggestion from your last comment

>     When creating solutions based on the Embedded C SDK you should consider :

Reviewer 4: ```suggestion
    When creating solutions based on the Embedded C SDK you should consider:
```

>     1. For production environments, the IoT Hub certificate must be validated during the TLS handshake.

Reviewer: ```suggestion
    1. The IoT Hub certificate must be validated during the TLS handshake for production environments.
```

Reviewer 2: Again, notice that now we have server cert validation on all our samples, even for prototyping. 
I don't think  we should create space for customers to work with our code in unsafe situation, if they do that they must make that choice themselves. LST, I would suggest removing "For production environment, ".

>     2. We use the Baltimore root CA to validate IoT Hub and Device Provisioning Service (DPS) server certificate (for the regions that use this certificate)

Reviewer: ```suggestion
```

>     2. We use the Baltimore root CA to validate IoT Hub and Device Provisioning Service (DPS) server certificate (for the regions that use this certificate)

Reviewer 4: ```suggestion
    2. We use the Baltimore root CA to validate IoT Hub and Device Provisioning Service (DPS) server certificate (for the regions that use this certificate).
```

>     3. For other regions (and private cloud environments) the appropriate root CA shall be used

Reviewer: ```suggestion
    2. Private cloud environments and regions that do not use the Baltimore root CA should use an appropriate root CA instead.
```

>     3. For other regions (and private cloud environments) the appropriate root CA shall be used

Reviewer 4: ```suggestion
    3. For other regions (and private cloud environments) the appropriate root CA shall be used.
```

>     ### Additional Information

Reviewer: ```suggestion
    #### Additional Information
```

    </p>
    </details>

## Certificates - Important to know

> The Azure IoT Hub certificates presented during TLS negotiation shall be always validated using the appropriate root CA certificate(s).

Reviewer 5: ```suggestion
The Azure IoT service certificates presented during TLS negotiation shall be always validated, on the device, using the appropriate trusted root CA certificate(s).
```

> For the Node MCU ESP8266 sample our script `generate_arduino_zip_library.sh` automatically downloads the root certificate used in the United States regions (Baltimore CA certificate) and add it to the sketch project.

Reviewer 5: ```suggestion
For the Node MCU ESP8266 sample, our script `generate_arduino_zip_library.sh` automatically downloads the root certificate used in the United States regions (Baltimore CA certificate) and adds it to the sketch project.
```

For other regions (and private cloud environments), please use the appropriate root CA certificate.

### Additional Information

> For additional guidance and important information about certificates, please refer to [this blog post](https://techcommunity.microsoft.com/t5/internet-of-things/azure-iot-tls-changes-are-coming-and-why-you-should-care/ba-p/1658456) from the security team. 

Reviewer 5: ```suggestion
For important information and additional guidance about certificates, please refer to [this blog post](https://techcommunity.microsoft.com/t5/internet-of-things/azure-iot-tls-changes-are-coming-and-why-you-should-care/ba-p/1658456) from the security team. 
```
